### DISCUSSION TITLE
Welcome to KaHIP Discussions!

### DISCUSSION THREAD

**Discussion by schulzchristian** [KaHIP/KaHIP#67] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mhoff** [comment]
Hello @schulzchristian, first of all I'd like to thank you (and your colleagues) for the hard work put into this library. I have started using KaHIP in a personal web project (https://github.com/mhoff/meshup) which uses KaFFPa as a WebAssembly build (https://github.com/mhoff/KaHIP-wasm) for a specific problem statement, which is surely not the primary focus of this library.

I am using KaFFPa on a potentially fully meshed, edge-weighted graph of people (=nodes) to generate subgroups (=partitions). The basic concept already works, however what I desire is more control over the individual sizes of the final partitions. Intuitive example: except for the trivial case n=1, there shall never be a partition with only one member (as the app is about people in a partition connecting with each other). In general, I would like the user to define a range of acceptable group sizes, around a primary group size which is always preferred. I realize that I probably need to implement that myself in or on top of KaFFPa. In the short run, I have already considered simply merging partitions with a single member into another partition. But I want to, as well as possible, retain (heuristic) optimality.

I would very much appreciate some hints on how to tackle that problem :)

Here is a productive deployment of the app: https://meshup.mhoff.net/
P.S. Of course KaHIP will be added as a reference in the imprint.

**Reply by schulzchristian** [reply]
Dear Michael, thank you using KaHIP! Lets discuss in a brief meeting, so lets schedule one. Could you send me an email to christian.schulz@informatik.uni-heidelberg.de? @DanielSeemaier do you also want to join the meeting?

Best
Christian

**Reply by DanielSeemaier** [reply]
Sure

**Reply by mhoff** [reply]
Many thanks! I have sent you both an email :)

**Comment by idubinkin** [comment]
Hello @schulzchristian.
Did i get it right that this software only allows to minimize number of edges between resulting subgraphs and not their summary weights?

**Reply by schulzchristian** [reply]
it minimizes the sum of the edge weights running between blocks (you call subgraphs). If the graph is unweighted, weights are assumed to be one, so it is the number of edges. If there are weights, it is really the sum of the weights that is minimized.

**Reply by idubinkin** [reply]
Thank you for clarification. 
Can it break a graph in a number of blocks that is not a power of 2?

**Comment by schulzchristian** [comment]
Yes, simply set the number of blocks k to the desired numver of blocks.

Best
Christian

**Reply by idubinkin** [reply]
Thank you.
